packages/bff/src/model/api/converters/catalogClientApiConverter.ts

```diff
@@ -123,24 +123,27 @@ export function toBffCatalogApiDescriptorInterface(
 export function toBffCatalogApiEserviceRiskAnalysis(
   riskAnalysis: catalogApi.EServiceRiskAnalysis
 ): bffApi.EServiceRiskAnalysis {
-  const answers: { [key: string]: string[] } = {};
-
-  riskAnalysis.riskAnalysisForm.singleAnswers
-    .concat(riskAnalysis.riskAnalysisForm.multiAnswers)
-    .map((answer) => ({
-      questionId: answer.questionId,
-      answer: answer.value,
-    }))
-    .forEach((QA) => {
-      if (answers[`${QA.questionId}`] && QA.answer) {
-        answers[`${QA.questionId}`] = [
-          ...answers[`${QA.questionId}`],
-          QA.answer,
-        ];
-      } else {
-        answers[`${QA.questionId}`] = [];
-      }
-    });
+  const answers: { [key: string]: string[] } =
+    riskAnalysis.riskAnalysisForm.singleAnswers
+      .concat(
+        riskAnalysis.riskAnalysisForm.multiAnswers.flatMap((multiAnswer) =>
+          multiAnswer.values.map((answerValue) => ({
+            id: multiAnswer.id,
+            value: answerValue,
+            key: multiAnswer.key,
+          }))
+        )
+      )
+      .reduce((answers: { [key: string]: string[] }, answer) => {
+        const key = `${answer.key}`;
+        if (answers[key] && answer.value) {
+          answers[key] = [...answers[key], answer.value];
+        } else {
+          answers[key] = [];
```

Comment:
We can use @ecamellini suggestion here too for answers' type and also we could remove `const key` and just use `answer.key` as the index, what do you think?

Reply:
🤔  IMHO
in this case for readability it could be more clear using a const `key` and made more coincise the subsequent  code.  For example without const there was `answer` overload

```
 if (answers[answer.key] && answer.value) {
          answers[answer.key] = [...answers[answer.key], answer.value];
        } else {
          answers[answer.key] = [];
        }
        return answers;
}, {});
```

meanwhile when I reading key it results more coincise and clear on first reading

```
 const key = `${answer.key}`;
 if (answers[key] && answer.value) {
          answers[key] = [...answers[key], answer.value];
        } else {
          answers[key] = [];
        }
        return answers;
}, {});
```